benchmark/buffers/buffer_zero.js

```diff
@@ -3,16 +3,18 @@
 const common = require('../common.js');
 
 const bench = common.createBenchmark(main, {
-  n: [1024]
+  n: [1024],
+  type: ['buffer', 'string']
 });
 
-const zero = Buffer.alloc(0);
+const zeroBuffer = Buffer.alloc(0);
+const zeroString = '';
 
 function main(conf) {
   var n = +conf.n;
   bench.start();
   for (let i = 0; i < n * 1024; i++) {
-    Buffer.from(zero);
+    conf.type === 'buffer' ? Buffer.from(zeroBuffer) : Buffer.from(zeroString);
```

Comment:
The change to lib/buffer.js LGTM but I wonder if it makes a difference on the benchmark numbers when you hoist out the conditional like this:
```js
if (conf.type === 'buffer')
  for (let i = 0; i < n * 1024; i++) Buffer.from(zeroBuffer);
else if (conf.type === 'string')
  for (let i = 0; i < n * 1024; i++) Buffer.from(zeroString);
```
Also, I'm not sure if crankshaft will constant-fold the expression `n * 1024` away, you might want to check if caching it makes a difference.